sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreter.scala

```diff
@@ -17,32 +17,42 @@
 
 package org.apache.spark.sql.scripting
 
-import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.{Row, SparkSession}
 import org.apache.spark.sql.catalyst.analysis.UnresolvedIdentifier
-import org.apache.spark.sql.catalyst.parser.{CompoundBody, CompoundPlanStatement, IfElseStatement, SingleStatement, WhileStatement}
-import org.apache.spark.sql.catalyst.plans.logical.{CreateVariable, DropVariable, LogicalPlan}
+import org.apache.spark.sql.catalyst.plans.logical.{CompoundBody, CompoundPlanStatement, CreateVariable, DropVariable, IfElseStatement, LogicalPlan, SingleStatement, WhileStatement}
 import org.apache.spark.sql.catalyst.trees.Origin
 
 /**
  * SQL scripting interpreter - builds SQL script execution plan.
+ *
+ * @param session
+ *   Spark session that SQL script is executed within.
  */
-case class SqlScriptingInterpreter() {
+case class SqlScriptingInterpreter(session: SparkSession) {
+
+  /**
+   * Execute the provided CompoundBody and return the result.
+   *
+   * @param compoundBody
+   *   CompoundBody to execute.
+   * @return Result of the execution.
```

Comment:
nit: It would be nice if you provide more details here like a sequence of rows as the result of execution of last statement in the given compound statement, or something like this.

Reply:
Done.